res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+          if (component.unshift) {
+            component.shift();
+          }
+        });
+      } else {
+        deckData.forEachComponent(function(component) {
+          if (component.unshift) {
+            component.unshift();
+          }
+        });
+      }
+    },
+  });
+
+  this.loadButton = new components.Button({
+    midi: [0x90 + midiChannel, 0x0D],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    unshift: function () {
+      this.inKey = 'LoadSelectedTrack';
+    },
+    shift: function () {
+      this.inKey = 'eject';
+    },
+  });
+
+  // Transport section
+  // Play button, for some reason the group is not correct on this one?
+  this.playButton = new components.PlayButton({
+    midi: [0x90 + midiChannel, 0x07],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    unshift: function () {
+      this.input = function (channel, control, value, status, group) {
+        if (value === 0x7F) {
+          if (engine.getValue(deckData.currentDeck, "play_latched")) {      //play_indicator play_latched
+            var deck = parseInt(deckData.currentDeck.charAt(8));
+            if (deckData.slowPauseSetState[deck - 1]) {
+              engine.brake(deck,
+                1,//((status & 0xF0) !=== 0x80 && value > 0),
+                54);
+            } else {
+              script.toggleControl(deckData.currentDeck, "play");
+            }
+          } else {
+            script.toggleControl(deckData.currentDeck, "play");
+          }
+        }
+      };
+    },
+    shift: function () {
+      this.input = function (_channel, _control, _value, _status, group) {
+        engine.setValue(deckData.currentDeck, "play_stutter", true);
+      };
+    },
+  });
+
+  this.cueButton = new components.CueButton({
+    midi: [0x90 + midiChannel, 0x06],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    shift: function () {
+      this.inKey = "start_play";
+    },
+  });
+
+  this.syncButton = new components.SyncButton({
+    midi: [0x90 + midiChannel, 0x05],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    shift: function () {
+      this.inKey = "sync_key";
+    },
+  });
+
+  this.pflButton = new components.Button({
+    midi: [0x90 + midiChannel, 0x0C],
+    type: components.Button.prototype.types.toggle,
+    key: 'pfl',
+  });
+
+  // Top controls
+  // Vinyl button
+  this.vinylButton = new components.Button({
+    midi: [0x90 + midiChannel, 0x03],
+    shiftOffset: 3,
+    shiftControl: false,
+    shiftChannel: true,
+    sendShifted: true,
+    unshift: function () {
+      this.input = function(channel, _control, value, status, group) {
+        if (value === 0x7F) {
+          var deck = parseInt(deckData.currentDeck.charAt(8)) - 1;
+          var new_status = !deckData.vinylButtonState[deck];

```

Comment:
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions).
```suggestion
          var deck = script.deckFromGroup(deckData.currentDeck);
          var new_status = !deckData.vinylButtonState[deck - 1];
```